Rubberduck.RibbonDispatcher/AbstractCOM/IAbstractDispatcher.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Runtime.InteropServices;
+
+using Microsoft.Office.Core;
+
+namespace Rubberduck.RibbonDispatcher.AbstractCOM {
+    /// <summary>TODO</summary>
+    [ComVisible(true)]
+    [CLSCompliant(true)]
+    [InterfaceType(ComInterfaceType.InterfaceIsDual)]
+    [Guid(RubberduckGuid.IRibbonViewModel)]
+    public interface IRibbonViewModel {
+        /// <summary>TODO</summary>
+        [DispId(DispIds.RibbonFactory)]
+        IRibbonFactory RibbonFactory { get; }
+
+        /// <summary>TODO</summary>
+        [DispId(DispIds.Invalidate)]
+        void Invalidate();
+        /// <summary>TODO</summary>
+        [DispId(DispIds.InvalidateControl)]
+        void InvalidateControl(string ControlId);
+        /// <summary>TODO</summary>
+        [DispId(DispIds.InvalidateControlMso)]
+        void InvalidateControlMso(string ControlId);
+        /// <summary>TODO</summary>
+        [DispId(DispIds.ActivateTab)]
+        void ActivateTab(string ControlId);
+
+        /// <summary>TODO</summary>
+        [DispId(DispIds.LoadImage)]
+        object LoadImage(string imageId);
+
+        /// <summary>Call back for GetDescription events from ribbon elements.</summary>
+        [DispId(DispIds.Description)]
+        string        GetDescription(IRibbonControl Control);
+        /// <summary>Call back for GetEnabled events from ribbon elements.</summary>
+        [DispId(DispIds.IsEnabled)]
+        bool          GetEnabled    (IRibbonControl Control);
+        /// <summary>Call back for GetKeyTip events from ribbon elements.</summary>
+        [DispId(DispIds.KeyTip)]
+        string        GetKeyTip     (IRibbonControl Control);
+        /// <summary>Call back for GetLabel events from ribbon elements.</summary>
+        [DispId(DispIds.Label)]
+        string        GetLabel      (IRibbonControl Control);
+        /// <summary>Call back for GetScreenTip events from ribbon elements.</summary>
```

Comment:
I really don't get this. You are demanding that everything look vanilla regardless of the message that must be communicated by the code. That even when the code blatantly violates DRY due to inadequacies of the language, and providing a means to, at a quick glance, ensure no violations of the required pattern exists, IS A REALLY GOOD THING TO PROVIDE;  the style guide still rules and these demand of the attentive reader must be buried. I am beginning to resent your insistence on blind adherence without providing any alternative solution to the problem being addressed by the violations. We all know that in-line code sucks, because it is immediately obsolete and unmaintainable in practice. so I attempt to communicate these messages by the structure of the code. I am open to alternative solutions, but not blind denial of the problem and challenge.